task6/script.js

```diff
@@ -0,0 +1,47 @@
+const df = document.createDocumentFragment();
+
+const ul = document.getElementById("ul");
+
+const listItem = [
+  {
+    to: "bookmark.html",
+    img: "1.png",
+    alt: "画像1",
+    text: "ブックマーク"
+  },
+  {
+    to: "message.html",
+    img: "2.png",
+    alt: "画像2",
+    text: "メッセージ"
+  }
+];
+
+const arrayPromise = () => {
+  return new Promise((resolve, reject) => {
+    resolve(listItem);
+  });
+};
+
+arrayPromise().then((value) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve();
+      for (let i = 0; i < value.length; i++) {
+        const li = document.createElement("li");
+        const a = document.createElement("a");
+        const img = document.createElement("img");
+
+        a.href = value[i].to;
+        a.textContent = value[i].text;
+
+        img.src = value[i].img;
+        img.alt = value[i].alt;
+
+        df.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
+      }
+      ul.appendChild(df);
+
+    }, 3000);
```

Comment:
[IMO]
Does this mean you have to wait 3 seconds for the DOM to be created?
I thought it would be better if you wait 3 seconds for the DOM to be created, after waiting for "Promise" to be "Resolved".

Reply:
Thank you for the review.
In this case, I found that I put the setTimeout method in the arraypromise object.

Do you understand that inserting it into an arraypromise object takes 3 seconds to resolve it, and inserting it into the then method takes 3 seconds to create a DOM?

Reply:
I'm sorry. My English sentence below was wrong.

What I meant to say was that the DOM generation should be done with the value resolved after 3 seconds.